packages/worker-shared/src/getUrlMetadata.ts

```diff
@@ -38,17 +44,25 @@ class IconExtractor {
 	}
 }
 
-export async function unfurl(url: string) {
+export async function getUrlMetadata({ url }: { url: string }) {
 	const meta$ = new MetaExtractor()
 	const title$ = new TextExtractor()
 	const icon$ = new IconExtractor()
+
+	let response
+	try {
+		response = await fetch(url)
+	} catch {
+		return null
+	}
+
 	// we use cloudflare's special html parser https://developers.cloudflare.com/workers/runtime-apis/html-rewriter/
 	await new HTMLRewriter()
 		.on('meta', meta$)
 		.on('title', title$)
 		.on('link', icon$)
-		.transform((await fetch(url)) as any)
-		.blob?.()
+		.transform(response as any)
+		.blob()
```

Comment:
i wonder whether we should budge this up into the try block too, in case of malformed html or non-html responses

Reply:
good call